src/main/java/seedu/address/logic/commands/wallet/WalletDeleteCommand.java

```diff
@@ -0,0 +1,58 @@
+package seedu.address.logic.commands.wallet;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliPrefix.WALLET_COMMAND_TYPE;
+
+import java.util.List;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.Command;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.transaction.Transaction;
+
+/**
+ * Deletes a transaction identified using it's displayed index from the wallet.
+ */
+public class WalletDeleteCommand extends Command {
+    public static final String COMMAND_WORD = "delete";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Deletes the expense/income identified by the index number used in the displayed transaction list.\n"
+            + "Parameters: INDEX (must be a positive integer)\n"
+            + "Example: " + WALLET_COMMAND_TYPE + " "
+            + COMMAND_WORD + " 1";
+
+    public static final String MESSAGE_DELETE_TRANSACTION_SUCCESS = "Deleted Income/Expense: %1$s";
+
+    private final Index targetIndex;
+
+    public WalletDeleteCommand(Index targetIndex) {
+        this.targetIndex = targetIndex;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        List<Transaction> lastShownList = model.getFilteredTransactionList();
+
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_TRANSACTION_DISPLAYED_INDEX);
+        }
+
+        Transaction transactionToDelete = lastShownList.get(targetIndex.getZeroBased());
+        model.deleteTransaction(transactionToDelete);
```

Comment:
transactionToDelete is of type Transaction. In your model manager, you checked: 
`if (transactionToDelete instanceof Expense) {
            deleteExpense((Expense) transactionToDelete);
        } else if (transactionToDelete instanceof Income) {
            deleteIncome((Income) transactionToDelete);
       }`

Not sure if instanceof will work, since it's of Type Transaction. Did you write and pass test for this? I'm not too sure also haha must try out

Reply:
Added more test, I believe instanceof subclass works as well!